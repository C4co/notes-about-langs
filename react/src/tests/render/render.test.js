import React from "react"
import { shallow, mount, render } from "enzyme"
import Mock from "./mock.component"
import Renderer from "react-test-renderer"

describe("Shallow Rendering", () => {
  test("Checking the mock (shallow)", () => {
    const wrapper = shallow(<Mock />)

    expect(wrapper.find(".wrapper").type()).toBe("section")
    expect(wrapper.find(".wrapper").children().length).toBe(2)

    wrapper.find(".title").forEach(node => {
      expect(node.hasClass("title")).toBe(true)
      expect(node.type()).toBe("h1")
      expect([" First Title ", " Second Title "].includes(node.text())).toBe(true)
    })
  })
})

describe("Mount rendering", () => {
  test("Checking the mock (mount)", () => {
    const wrapper = mount(<Mock />)
    expect(wrapper.find("h1").length).toBe(2)
  })
})

describe("Static render", () => {
  test("Checking the mock (static)", () => {
    const wrapper = render(<Mock />)
    expect(wrapper.find("h1").length).toBe(2)
  })
})

describe("Working with react-test-renderer", () => {
  test("Testing with react-test-renderer", () => {
    const component = Renderer.create(<Mock/>)
    expect(component).toMatchSnapshot()
  })
})
