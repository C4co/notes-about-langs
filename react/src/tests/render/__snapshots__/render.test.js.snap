// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Working with react-test-renderer Testing with react-test-renderer 1`] = `
<section
  className="wrapper"
>
  <h1
    className="title"
    id="title1"
  >
     First Title 
  </h1>
  <h1
    className="title"
    id="title2"
  >
     Second Title 
  </h1>
</section>
`;
